### PR TITLE
Carry #14813 on linting package opts and trust

### PR DESCRIPTION
**Pull request by vdemeester**
Carry the work of @sevki, updated according to comments and rewrote a bit some on opts (that were there before #14756).

/cc @LK4D4 @dmcgowan @tiborvass 
Closes #14813.

### CODE REVIEW

File: trust/trusts.go [hunk 1]
```diff
@@ -38,7 +41,9 @@ const defaultFetchtime = 45 * time.Second
 
 var baseEndpoints = map[string]string{"official": "https://dvjy3tqbc323p.cloudfront.net/trust/official.json"}
 
-func NewTrustStore(path string) (*TrustStore, error) {
+// NewTrustStore creates a TrustStore from a given path, if the path is not
+// relative, it will be joined with the working directory.
+func NewTrustStore(path string) (*Store, error) {
```
Comment (cpuguy83) [inline trust/trusts.go]: Surprised the linter didn't pick this one up, but I think the function name should be `NewStore`

File: opts/ip.go [hunk 2]
```diff
@@ -5,20 +5,25 @@ import (
 	"net"
 )
 
-// IpOpt type that hold an IP
-type IpOpt struct {
+// IPOpt holds an IP
```
Comment (cpuguy83) [inline opts/ip.go]: But why does it hold an IP? What is it for?

I think generally for all these types we have we can probably just add that these are used for storing values from CLI flags.

File: trust/trusts.go (hunk 1 above)
Comment (vdemeester) [inline trust/trusts.go]: Ah right, true :sweat_smile:

**Comment by LK4D4** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

**Pull request merged**